R/mz_summary.R

```diff
@@ -0,0 +1,135 @@
+#' Also the plots will be displayed. 
+#' The default is filter = F, so the unfiltered summary will be returned. 
+#' If filter is set to a `number`, e.g. 1000 only mz differences above 
+#' this threshold will be displayed. 
+#' The function can be applied for adjacency lists from `structural` and `combine`
+#' 
+#' @return 
+#' `data.frame` containing the numbers of present mz differences and
+#' corresponding name. Also a plot will be displayed.
+#'
+#' @author Liesa Salzer, \email{liesa.salzer@@helmholtz-muenchen.de}
+#'
+#' @examples
+#' data("x_test", package = "MetNet")
+#' transformation <- rbind(
+#'     c("Monosaccharide (-H2O)", "C6H10O5", "162.0528234315"),
+#'     c("Disaccharide (-H2O)", "C12H20O11", "340.1005614851"),
+#'     c("Trisaccharide (-H2O)", "C18H30O15", "486.1584702945"))
+#' transformation <- data.frame(group = transformation[, 1],
+#'                                 formula = transformation[, 2],
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' mz_summary(am_struct)
+#'
+#'
+#'@export
+#'
+#'@importFrom magrittr %>%
+mz_summary <- function(x, filter = F, ...){
+  
+  if (!"AdjacencyMatrix" %in% class(x)) 
+    stop("x is not class AdjacencyMatrix")
+  
+  x_df <- as.data.frame(x)
+
+  if (!"mass_difference" %in% colnames(x_df)) 
+    stop("x does not contain a mass-difference adjacency matrix")
+  
+  if (!1 %in% x_df$binary) 
+    stop("x does not contain any mass-differences")
+  
+  # if AdjacencyMatrix from `combine` is used 
+  if("combine_mass_difference" %in% colnames(x_df)){
+    x_df <- x_df[x_df$combine_binary == 1, ]
+    x_df <- na.omit(x_df, "combine_binary")
+    
+    sum_mass <- x_df %>% 
+      dplyr::group_by(`combine_mass_difference`) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- x_df %>% 
+      dplyr::group_by(`combine_transformation`) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$`combine_mass_difference`)
+  }
+  
+  # if AdjacencyMatrix from `structural` is used 
+  else {
+    x_df <- x_df[x_df$binary == 1, ]
+    
+    sum_mass <- x_df %>% 
+      dplyr::group_by(`mass_difference`) %>%  
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()
+    
+    sum_transform <- x_df %>% 
+      dplyr::group_by(`transformation`) %>% 
+      dplyr::summarise(count = dplyr::n()) %>%
+      as.data.frame()  %>% 
+      tibble::add_column(sum_mass$`mass_difference`)
+  }
+  
+    colnames(sum_transform) <- c("transformation", "counts", "mass_difference")
+    sum_transform <- sum_transform %>% 
+      dplyr::select(transformation, `mass_difference`, counts)
+  
+  if (filter == F) {
+    plot_list <- ggplot2::ggplot(sum_transform, 
+                                 ggplot2::aes(x=transformation, y=counts)) + 
```

Comment:
could you please use `aes_string` instead of `aes`? for me the check will complain if there are undefined variables in the code.